File: Daily Challenge/March 2025/Divide Array Into Equal Pairs.cs
public class Solution {
  public bool DivideArray(int[] nums) {
    Dictionary<int, int> counts = new Dictionary<int, int>();

    foreach (int num in nums) {
      if (!counts.ContainsKey(num)) {
        counts[num] = 0;
      }

      counts[num]++;
    }

    foreach (int count in counts.Values) {
      if (count % 2 != 0) {
        return false;
      }
    }

    return true;
  }
}

File: Daily Challenge/March 2025/Divide Array Into Equal Pairs.cpp
class Solution {
public:
  bool divideArray(vector<int>& nums) {
    map<int, int> count;
    for (int num : nums) {
      count[num]++;
    }

    for (auto& [num, freq] : count) {
      if (freq % 2 != 0) {
        return false;
      }
    }

    return true;
  }
};